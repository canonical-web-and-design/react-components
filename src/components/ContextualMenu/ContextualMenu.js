import React from "react";
import PropTypes from "prop-types";

const ContextualMenu = ({ children, visible = false }) => (
  <div>
    <span className="p-contextual-menu--left">
      <span
        className="p-contextual-menu__dropdown"
        aria-hidden={!visible}
        aria-label="submenu"
      >
        {children}
      </span>
    </span>
  </div>
);

ContextualMenu.propTypes = {
  visible: PropTypes.bool,
  children: PropTypes.any,
};

export default ContextualMenu;
