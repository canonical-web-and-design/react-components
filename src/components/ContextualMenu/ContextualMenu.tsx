import React from "react";
import type { ReactNode } from "react";
import PropTypes from "prop-types";

type Props = {
  visible?: boolean;
  children?: ReactNode;
};

const ContextualMenu = ({ children, visible = false }: Props): JSX.Element => (
  <div>
    <span className="p-contextual-menu--left">
      <span
        className="p-contextual-menu__dropdown"
        aria-hidden={!visible}
        aria-label="submenu"
      >
        {children}
      </span>
    </span>
  </div>
);

ContextualMenu.propTypes = {
  visible: PropTypes.bool,
  children: PropTypes.any,
};

export default ContextualMenu;
