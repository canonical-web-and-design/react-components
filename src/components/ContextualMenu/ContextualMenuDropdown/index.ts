export { default } from "./ContextualMenuDropdown";
export type { MenuLink, Position } from "./ContextualMenuDropdown";
