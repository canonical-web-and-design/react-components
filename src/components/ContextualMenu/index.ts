export { default } from "./ContextualMenu";
export type { Props } from "./ContextualMenu";
export type { MenuLink, Position } from "./ContextualMenuDropdown";
