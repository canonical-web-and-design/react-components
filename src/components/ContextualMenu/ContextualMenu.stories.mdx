import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs/blocks";
import { withKnobs } from "@storybook/addon-knobs";
import ContextualMenu from "./ContextualMenu";

<Meta
  title="ContextualMenu"
  component={ContextualMenu}
  decorators={[withKnobs]}
/>

### Contextual Menu

This is a [React](https://reactjs.org/) component for the Vanilla [Contextual menu](https://docs.vanillaframework.io/patterns//contextual-menu/).

A contextual menu can be used in conjunction with any page element to provide a contextual menu.

### Props

<ArgsTable of={ContextualMenu} />

### Default

The contextual menu will provide a visual wrapper to any provided children. Visbility can be toggled via the `visible` prop.

<Canvas>
  <Story name="Default">
    <ContextualMenu visible={true}>
      <a href="#">Link 1</a>
      <a href="#">Link 2</a>
      <a href="#">Link 3</a>
    </ContextualMenu>
  </Story>
</Canvas>
