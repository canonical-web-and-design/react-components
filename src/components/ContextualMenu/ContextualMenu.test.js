import { shallow } from "enzyme";
import React from "react";

import ContextualMenu from "./ContextualMenu";

describe("Contextual Menu ", () => {
  it("renders instance but not visible", () => {
    const wrapper = shallow(<ContextualMenu>Test content</ContextualMenu>);
    expect(
      wrapper.find(".p-contextual-menu__dropdown").prop("aria-hidden")
    ).toEqual(true);
  });

  it("renders visible instance", () => {
    const wrapper = shallow(
      <ContextualMenu visible={true}>Test content</ContextualMenu>
    );
    expect(
      wrapper.find(".p-contextual-menu__dropdown").prop("aria-hidden")
    ).toEqual(false);
  });
});
