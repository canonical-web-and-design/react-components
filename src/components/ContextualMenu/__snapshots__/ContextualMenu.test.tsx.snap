// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextualMenu  renders 1`] = `
<ContextualMenu
  links={Array []}
>
  <span
    className="p-contextual-menu"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <span
      style={
        Object {
          "display": "none",
        }
      }
    />
  </span>
</ContextualMenu>
`;
